models/DataObject/ClassDefinition/Data/QuantityValue.php

```diff
@@ -402,12 +662,40 @@ public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
             throw new Model\Element\ValidationException('Empty mandatory field [ '.$this->getName().' ]');
         }
 
+        if ($data !== null && !$this->isEmpty($data->getValue()) && !is_numeric($data->getValue())) {
```

Comment:
just wondering, would be there any drawbacks by moving this block inside the one line 676 since it shares 2 of the conditions?

Reply:
Not sure. Numeric uses the same logic
https://github.com/pimcore/pimcore/blob/10.x/models/DataObject/ClassDefinition/Data/Numeric.php#L511

Reply:
ah, i see, there's a difference on how omitMandatoryCheck is managed.
IMHO the best would be that both be almost the same, by even removing this

if ($omitMandatoryCheck) {
    return;
 }
and adding the omitMandatoryCheck to the IF block in the same way as numeric, otherwise it would be inconsistent.
Could you please do the adaptations?

Reply:
Done :)